apps/angular/ngfor-enhancement/src/app/common/directives/ngForEmpty.directive.ts

```diff
@@ -0,0 +1,37 @@
+import { NgFor } from '@angular/common';
+import {
+  Directive,
+  DoCheck,
+  EmbeddedViewRef,
+  Input,
+  TemplateRef,
+  ViewContainerRef,
+  inject,
+} from '@angular/core';
+
+@Directive({
+  selector: '[ngForEmpty]',
```

Comment:
you could have used ngFor as a selector as well. I can simplify the code because you don't have to map ngForOf.

Reply:
yes but i like this this way, it reminds me there is a optional else block which replaces a if else statement